controllers/profiles/common/app_properties.go

```diff
@@ -267,6 +268,35 @@ func generateDiscoveryProperties(ctx context.Context, catalog discovery.ServiceC
 			}
 		}
 	}
+
+	for _, function := range workflow.Spec.Flow.Functions {
```

Comment:
This method is getting quite large. Maybe a little refactoring here to an `app_properties_discovery.go` file that would wrap al of this? wdyt?